src/istore.c

```diff
@@ -1096,6 +1096,58 @@ Datum istore_slice_to_array(PG_FUNCTION_ARGS)
     PG_RETURN_POINTER(aout);
 }
 
+static void istore_clamp_pass(IStore * is, int32 end_key, int delta_dir)
+{
+    IStorePair * pairs;
+    int32 result = 0;
+    int   index  = 0, count = 0;
+
+    pairs = FIRST_PAIR(is, IStorePair);
+    index = delta_dir > 0 ? 0 : is->len - 1;
+    while ( ((delta_dir > 0) && (index < is->len && pairs[index].key <= end_key)) ||
+            ((delta_dir < 0) && (index >= 0      && pairs[index].key >= end_key)) )
+    {
+        is->buflen -= is_pair_buf_len(pairs + index);
+        result = DirectFunctionCall2(int4pl, result, pairs[index].val);
```

Comment:
`int4pl` it rather short all it does is overflow checking and I guess that a `DirectFunctionCall2` is potentially expensive. Maybe it's worth using [INTPL](https://github.com/adjust/istore/blob/master/src/istore.h#L205)

Reply:
The only other place I found that uses this macro is `(big)istore_avl_transfn` everything else e.g. `istore_accumulate` uses `DirectFunctionCall2` (that's why I also went this way). I don't have an opinion on the matter, but I'm curious why it was written that way.

Reply:
yep this performance optimization was introduced with 8cc0510bc63ecee11ad8772ace1eb392446377e5

Reply:
the `DirectFunctionCall2` was used to dryout code through a functional approach in `istore_apply_datum`

Reply:
Thanks for the explanation.